### PR TITLE
docs(backup): add open api spec for list backup api

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR adds an open api spec for list backup api. List operation is not implemented yet. The goal is to define the spec now, and the implementation would follow this spec.

You can list all backup via `GET  actuator/backups/`. It returns a list of backups with detailed info of each backup. ([Ref](https://docs.google.com/document/d/1pITW-pvUiGuJ8kIeR1vqjnoqmhs0unnNN955jUkhF6w/edit#heading=h.8pku8i8evwlw)) 

The other operations (take, get and delete) will be added later to the same spec.

You can visualize the spec online using https://editor.swagger.io/.

## Related issues

related to #10712

### CODE REVIEW

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          readOnly: true
+          type: string
+          format: date-time
+          example: "2022-09-15T13:10:38.176514094Z"
+        descriptor:
+          readOnly: true
+          allOf:
+            - $ref: '#/components/schemas/BackupDescriptor'
+      required:
+        - id
```
Comment (oleschoenburg) [inline docs/actuator-api-spec/backup-management-api.yaml:151]: :x: 
```suggestion
        - partitionId
```

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        details:
+          readOnly: true
+          description: |
+            Detailed list of the status of the backup per partition. It should always contain all
+            partitions known to the cluster.
+          type: array
+          items:
+            allOf:
+              - $ref: '#/components/schemas/PartitionBackupInfo'
+      required:
```
Comment (oleschoenburg) [inline docs/actuator-api-spec/backup-management-api.yaml:114]: :wrench: I suspect that we will always return details
```suggestion
      required:
        - details
```

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        - If one is 'FAILED', then the overall state is 'FAILED'.
+        - Otherwise, if one is 'DOES_NOT_EXIST', then the overall state is 'INCOMPLETE'.
+        - Otherwise, if one is 'IN_PROGRESS', then the overall state is 'IN_PROGRESS'.
+      type: object
+      properties:
+        backupId:
+          readOnly: true
+          allOf:
+            - $ref: '#/components/schemas/BackupId'
+        state:
```
Comment (oleschoenburg) [inline docs/actuator-api-spec/backup-management-api.yaml:97]: :question: 
We usually called this `statusCode` not `state`. Is there a reason for this difference?

**Review by lenaschoenburg — CHANGES_REQUESTED** [review]
Thanks @deepthidevaki, I think I've found a few bugs and marked some inconsistencies with our existing API. Please take a look.

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    StateCode:
+      title: Backup State
+      description: The state of the backup.
+      type: string
+      enum:
+        - DOES_NOT_EXIST
+        - IN_PROGRESS
+        - COMPLETED
+        - FAILED
+        - INCOMPLETE
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply docs/actuator-api-spec/backup-management-api.yaml:73]: This is new, to align with Operate and Optimize api.

File: docs/actuator-api-spec/backup-management-api.yaml (hunk 3 above)
Comment (deepthidevaki) [reply docs/actuator-api-spec/backup-management-api.yaml:97]: This is to align with the Operate and Optimize api. We will change the implementation to reflect this spec.

File: docs/actuator-api-spec/backup-management-api.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply docs/actuator-api-spec/backup-management-api.yaml:73]: It is not yet in the implementation, but we will add it.

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        default: http
+        description: Management server schema
+paths:
+  /:
+    get:
+      summary: Lists all available backups
+      description: |
+        Returns a list of all available backups with their state and additional info,
+        sorted in descending order of backupId.
+      responses:
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply docs/actuator-api-spec/backup-management-api.yaml:25]: I think 501 Not Implemeted is better suited for that case. I will sync with Operate/Optimize before adding this.

File: docs/actuator-api-spec/backup-management-api.yaml [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+    PartitionId:
+      title: Partition ID
+      description: |
+        The ID of a partition. This is always a positive number greater than or equal to 1.
+      type: number
+      minimum: 1
+      example: 3
+
+    StateCode:
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply docs/actuator-api-spec/backup-management-api.yaml:64]: Operate and Optimize is using "state". So to align with them, we rename it to state.

**Review by lenaschoenburg — APPROVED** [review]
:+1:

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**